src/python/pants/option/global_options.py

```diff
@@ -703,6 +706,22 @@ class BootstrapOptions:
             """
         ),
     )
+    plugins_lockfile = StrOption(
+        advanced=True,
+        default="<none>",  # i.e. `NO_TOOL_LOCKFILE`, but avoid circular import.
```

Comment:
IMO, we should set a default here that will be used _iff_ there are plugins declared. While not everyone will be able to use the default, the alternative is that following a best practice _requires_ additional boilerplate, and that's not great.